db/notifications.go

```diff
@@ -0,0 +1,81 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/flanksource/duty/models"
+	"github.com/flanksource/incident-commander/api"
+	v1 "github.com/flanksource/incident-commander/api/v1"
+	"github.com/google/uuid"
+)
+
+func PersistNotificationFromCRD(obj *v1.Notification) error {
+	uid, err := uuid.Parse(string(obj.GetUID()))
+	if err != nil {
+		return err
+	}
+
+	if obj.Spec.To.Empty() {
+		return fmt.Errorf("notification %s has no recipient", obj.Name)
+	}
+
+	dbObj := models.Notification{
+		ID:         uid,
+		Events:     obj.Spec.Events,
+		Title:      obj.Spec.Title,
+		Template:   obj.Spec.Template,
+		Filter:     obj.Spec.Filter,
+		Properties: obj.Spec.Properties,
+	}
+
+	switch {
+	case obj.Spec.To.Person != "":
+		if uid, err := uuid.Parse(obj.Spec.To.Person); err == nil {
+			dbObj.PersonID = &uid
+		} else {
+			var person models.Person
+			if err := Gorm.Where("email = ?", obj.Spec.To.Person).First(&person).Error; err != nil {
+				return err
+			}
+			dbObj.PersonID = &person.ID
+		}
```

Comment:
Can we move these into a FindPerson/Team?  - I also think the person lookup should be uuid, name or email and our docs to use Name so as not confuse with to.email which doesn't require a person object

Reply:
>  I also think the person lookup should be uuid, name or email

`people.name` is not unique. Should we add that constraint?